agent/consul/rpc.go

```diff
@@ -351,11 +351,6 @@ func (s *Server) blockingQuery(queryOpts *structs.QueryOptions, queryMeta *struc
 	fn queryFn) error {
 	var timeout *time.Timer
 
-	// Fast path right to the non-blocking query.
```

Comment:
Previously, MaxQueryTime and MinQueryIndex had to be specified together. This short circuit seemed like unnecessary optimization. I've removed it so that the timer in line 365 can be initialized properly even when MinQueryIndex is zero.